### PR TITLE
AV-212214 Add URI encoding to rest calls

### PR DESCRIPTION
**Pull request by aaha97**
Add URI encoding to rest calls to support resource names in AVI 30.x

### CODE REVIEW

**Comment by aaha97** [comment]
build ako

File: tests/scaletest/lib/utils.go [hunk 1]
```diff
@@ -53,6 +53,10 @@ func (e *errorString) Error() string {
 	return e.s
 }
 
+func GetUriEncoded(uri string) string {
+	return strings.ReplaceAll(uri, " ", "%20")
```
Comment (ashishnayak96) [inline tests/scaletest/lib/utils.go:57]: can we look into url.PathEscape or url.QueryEscape to take care of special characters as well. This has been mentioned in the JIRA ticket to encode.

Comment (aaha97) [reply tests/scaletest/lib/utils.go]: updated code to use url.QueryEscape

**Comment by aaha97** [comment]
build ako

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -624,3 +624,10 @@ func String(s *string) string {
 	}
 	return ""
 }
+
+func GetUriEncoded(uri string) string {
+	if uriSplit := strings.SplitN(uri, "?", 2); len(uriSplit) == 2 {
+		return uriSplit[0] + "?" + url.QueryEscape(uriSplit[1])
```
Comment (ashishnayak96) [inline pkg/utils/utils.go:630]: shouldnt we PathEscape the first part of URL?

Comment (aaha97) [reply pkg/utils/utils.go:630]: shouldn't be required as we are making these calls to controller and the paths are known

**Comment by aaha97** [comment]
build ako

**Comment by aaha97** [comment]
build ako

**Review by ashishnayak96 — DISMISSED** [review]
LGTM

**Comment by aaha97** [comment]
build ako

**Review by arihantg — DISMISSED** [review]
LGTM

**Comment by aaha97** [comment]
build ako

**Comment by aaha97** [comment]
build ako

**Comment by aaha97** [comment]
adding UT results as sanity checks seem to be failing

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/k8stest       32.035s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/integrationtest       485.448s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/ingresstests  1014.532s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/oshiftroutetests      1314.515s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/bootuptests   21.533s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/multicloudtests       171.619s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/servicesapitests      80.110s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/advl4tests    47.849s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/evhtests      1687.716s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/evhtests      1593.148s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/dedicatedevhtests     693.625s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/dedicatedevhtests     789.622s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/dedicatedvstests      188.597s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/infratests    2.088s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/hatests       35.007s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/cnitests      104.666s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/cnitests      28.535s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/gatewayapitests/status        316.998s

ok      github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/multitenancytests     71.739s

**Review by ashishnayak96 — APPROVED** [review]
LGTM

**Review by arihantg — APPROVED** [review]

**Pull request merged**